deployment/shrink-image.sh

```diff
@@ -0,0 +1,218 @@
+#  may correct your OS disk with invalid input!!
+#
+
+# sd card device
+sd_devpath=$1
+
+# output image file
+output_img=$2
+
+# check for empty inputs
+if [ "x$sd_devpath" == "x" ] || [ "x$output_img" == "x" ]; then
+  echo "Usage: $0 /dev/sdb connectbox.img"
+  exit 1
+fi
+
+# Make sure we have root
+if [[ $EUID -ne 0 ]]; then
+   echo "This script must be run as root (or via sudo)" 1>&2
+   exit 1
+fi
+
+# check for block device
+if [ ! -b $sd_devpath ]; then
+  echo "$sd_devpath is not a valid device!"
+  exit -1
+fi
+
+# check for valid output directory
+output_dir=$(dirname $output_img)
+if [ ! -d $output_dir ]; then
+  echo "$output_dir is not a valid directory"
+  exit -1
+fi
+
+# make sure we won't overwrite an existing image
+if [ -f $output_img ]; then
+  echo "$output_img already exists!"
+  exit -1
+fi
+
+# make sure device is removable
+sd_blkdev=$(basename $sd_devpath)
+sd_sysfs=/sys/block/${sd_blkdev}
+sd_removable=$(cat $sd_sysfs/removable)
+if [ $sd_removable != 1 ]; then
+  echo "$sd_devpath is not a removable device (may not be an sd card)"
+  exit -1
+fi
+
+# make sure device isn't already mounted
+# check anything within the device; nothing should be
+sd_mounts=$(mount | grep -c "^$sd_devpath")
+if [ $sd_mounts -gt 0 ]; then
+  echo "$sd_devpath is already mounted!"
+  exit -1
+fi
+
+# count the device partitions
+sd_partcount=$(ls -d ${sd_sysfs}/${sd_blkdev}* | wc -l | tr -d '[:space:]')
+if [ $sd_partcount -le 0 ] || [ $sd_partcount -gt 2 ]; then
+  echo "$sd_devpath has $sd_partcount partitions (expected 1 or two)"
+  exit -1
+fi
+
+# if we have one partition, it's root:
+if [ $sd_partcount -eq 1 ]; then
+  boot_partition_num=0
+  boot_sysfs=""
+  boot_dev=""
+
+  root_partition_num=1
+  root_sysfs=${sd_sysfs}/${sd_blkdev}${root_partition_num}
+  root_dev=${sd_devpath}${root_partition_num}
+fi
+
+# if we have two partitions, they are probably boot & root:
+if [ $sd_partcount -eq 2 ]; then
+  boot_partition_num=1
+  boot_sysfs=${sd_sysfs}/${sd_blkdev}${boot_partition_num}
+  boot_dev=${sd_devpath}${boot_partition_num}
+
+  root_partition_num=2
+  root_sysfs=${sd_sysfs}/${sd_blkdev}${root_partition_num}
+  root_dev=${sd_devpath}${root_partition_num}
+fi
+
+# make a random temporary path to work in
+tmp_path=""
+while [ "x$tmp_path" == "x" ]; do
+  tmp_path=/tmp/connectbox.$RANDOM
+  if [ -d $tmp_path ]; then
+    tmp_path=""
+  fi
+done
+
+# sanity check since an empty path can be dangerous later
+if [ "x$tmp_path" == "x" ]; then
+  echo "Couldn't generate a unique tmp directory name!"
+  exit -1
+fi
+
+# create temporary path
+mkdir $tmp_path
+
+# make sure it exists...
+if [ ! -d $tmp_path ]; then
+  echo "Unable to create directory $tmp_path!"
+  exit -1
+fi
+
+# mount the boot fs and do cleanup
+if [ "x$boot_dev" != "x" ]; then
+  echo "preparing boot device $boot_dev"
+  boot_mount=${tmp_path}/boot
+  mkdir -p $boot_mount
+  mount $boot_dev $boot_mount
+  
+  # Raspbian: make the filesystem expand at the next boot
```

Comment:
No, because only Raspbian has a /boot partition, so it will only take effect for Armbian. Armbian does the expansion some other way.